### PR TITLE
Final submission : 100 % coverage 

### PR DESCRIPTION
**Pull request by roelof1967**
Hello, 

Lab says there is still no coverage on line 16 of index.js. 
I do not agree with it because I wrote two test which test if a server starts up on port 8000 (standard) and on port 5000. And the test are passed.

Anyone a tip how to solve this ?

### CODE REVIEW

File: test/index.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+var lab = exports.lab = Lab.script();
+var describe = lab.experiment;
+var expect = Code.expect;
+var it = lab.test;
+
+describe('Server', function() {
+   
+    it('starts server on port 8000 by default', function (done) {
+
+        Server.init(8000, function (err, server) {
```
Comment (EvanMarkHopkins) [inline test/index.js]: 8000 satisfies the first condition, try passing null or 0 instead.

**Comment by roelof1967** [comment]
@EvanMarkHopkins  Thanks, one small change and 100 % coverage. Now hope that the experts are happy too.

File: lib/index.js [hunk 2]
```diff
@@ -10,19 +10,25 @@ var Version = require('./version');
 var internals = {};
 
 
-internals.init = function () {
+exports.init = function (data, callback) {
```
Comment (chyld) [inline lib/index.js]: the variable name data is generic, use port

**Comment by roelof1967** [comment]
@chyld  thanks for the feedback, changed it already

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,10 @@
+language: node_js
+
+node_js:
+  - 0.10
+  - 0.12
+  - iojs
+
```
Comment (idanwe) [inline .travis.yml]: Remove empty lines

File: lib/index.js [hunk 4]
```diff
@@ -10,19 +10,25 @@ var Version = require('./version');
...
+        }
+
         server.start(function (err) {
 
-            Hoek.assert(!err, err);
-            console.log('Server started at: ' + server.info.uri);
-        });
-    });
-};
+           return callback(err, server);
```
Comment (idanwe) [inline lib/index.js]: Fix the indentations and remove the empty lines after `});`

File: lib/index.js [hunk 5]
```diff
@@ -10,19 +10,25 @@ var Version = require('./version');
 var internals = {};
 
 
-internals.init = function () {
+exports.init = function (port, callback) {
```
Comment (roelof1967) [inline lib/index.js]: oke, 
Can you explain what you mean here ?

Comment (idanwe) [inline lib/index.js]: There is two possibilities for the function arguments, `(callback)` or `(port, callback)`

And there are several solutions for this:
 * Type checking of the arguments - you can look at [my PR](https://github.com/idanwe/hueniversity/blob/assignment3/lib/index.js#L16-L19)
 * Arguments length checking - [hapi code](https://github.com/hapijs/hapi/blob/master/lib/server.js#L205-L206)

Comment (roelof1967) [inline lib/index.js]: Thanks, but as soon as I added this : 

~~~

if (typeof port === 'function') {
        callback = port;
        port = internals.port;
    }

~~~  

Then all the test of index are failing. I have then a coverage of 93%.

File: lib/index.js [hunk 6]
```diff
@@ -4,25 +4,29 @@ var Hapi = require('hapi');
...
 
-var internals = {};
+var internals = {
+    port: 8000
+};
 
+exports.init = function (port, callback) {
 
-internals.init = function () {
+    if (typeof port === 'function') {
```
Comment (hueniverse) [inline lib/index.js]: Not in the assignment.

File: start.js [hunk 7]
```diff
@@ -0,0 +1,9 @@
+var Hoek = require('hoek');
+
+var lib = require('./lib');
```
Comment (hueniverse) [inline start.js]: `Lib`

File: lib/index.js (hunk 6 above)
Comment (roelof1967) [inline lib/index.js]: oke, a few remarks before you idanwe told me to change it to this. Is this then a good solution :  port = port || 0 ;

File: start.js (hunk 7 above)
Comment (roelof1967) [inline start.js]: I will change this

Comment (idanwe) [inline start.js]: Sorry for the confusion

File: lib/index.js (hunk 6 above)
Comment (AdriVanHoudt) [inline lib/index.js]: Hapi will do that for you ;)

File: start.js [hunk 8]
```diff
@@ -0,0 +1,9 @@
+var Hoek = require('hoek');
+
+var Lib = require('./lib');
+
+lib.init(8000, function (err, server) {
```
Comment (arb) [inline start.js]: You sure this works?

File: .travis.yml [hunk 9]
```diff
@@ -0,0 +1,6 @@
+language: node_js
+
+node_js:
+  - 0.10
+  - 0.12
+  - iojs
```
Comment (arb) [inline .travis.yml]: Missing newline.

File: start.js (hunk 8 above)
Comment (roelof1967) [inline start.js]: Yep. all the test I made were green.

File: .travis.yml (hunk 9 above)
Comment (roelof1967) [inline .travis.yml]: Where does the newline to be placed ?

Comment (AdriVanHoudt) [inline .travis.yml]: at the end of the file

**Pull request closed without merging**